chips/nrf52840/src/ieee802154_radio.rs

```diff
@@ -1342,3 +1395,26 @@ impl<'a> kernel::hil::radio::RadioData<'a> for Radio<'a> {
         Ok(())
     }
 }
+
+impl DeferredCallClient for Radio<'_> {
+    fn handle_deferred_call(&self) {
+        // On deferred call we trigger the config or power callbacks. The
+        // `.take()` ensures we clear what is pending.
+        self.deferred_call_operation.take().map(|op| match op {
+            DeferredOperation::ConfigClientCallback => {
+                self.config_client.map(|client| {
+                    client.config_done(Ok(()));
+                });
+            }
+            DeferredOperation::PowerClientCallback => {
+                self.power_client.map(|client| {
+                    client.changed(self.radio_is_on());
+                });
+            }
+        });
+    }
+
+    fn register(&'static self) {
+        self.deferred_call.register(self);
+    }
+}
```

Comment:
I am likely missing something here, but can you elaborate further on how these deferred calls we are adding to the radio are intended to work.

Reply:
The deferred calls are used to call the ConfigClient and PowerClient callbacks from an interrupt (which is required for Tock's threading model). Since the nrf driver can do the operations in a synchronous manner, we use deferred calls to mimic an async operation.